iroh/src/magicsock/remote_map/remote_state.rs

```diff
@@ -278,29 +268,31 @@ impl RemoteStateActor {
                 }
             }
 
-            if self.connections.is_empty() && inbox.is_idle() && idle_timeout.is_none() {
+            let is_idle = self.connections.is_empty() && inbox.is_idle();
+            if idle_timeout.is_none() && is_idle {
                 trace!("start idle timeout");
                 idle_timeout
                     .as_mut()
                     .set_future(time::sleep(ACTOR_MAX_IDLE_TIMEOUT));
-            } else if idle_timeout.is_some() {
+            } else if idle_timeout.is_some() && !is_idle {
                 trace!("abort idle timeout");
                 idle_timeout.as_mut().set_none()
             }
         }
         trace!("actor terminating");
-        Ok(())
     }
 
     /// Handles an actor message.
     ///
     /// Error returns are fatal and kill the actor.
     #[instrument(skip(self))]
-    async fn handle_message(&mut self, msg: RemoteStateMessage) -> n0_error::Result<()> {
+    async fn handle_message(&mut self, msg: RemoteStateMessage) {
         // trace!("handling message");
         match msg {
             RemoteStateMessage::SendDatagram(transmit) => {
-                self.handle_msg_send_datagram(transmit).await?;
+                if let Err(err) = self.handle_msg_send_datagram(transmit).await {
+                    warn!("failed to send datagram: {err:#}");
+                }
```

Comment:
I think this is a bug:
We should avoid short-circuiting not on the `handle_msg_send_datagram` level, but on the `send_datagram` level. Otherwise a single path that fails will make us stop sending datagrams to any other paths that might still work.

So in essence that just means this `if let Err(err) = ...` should be moved inside the `handle_msg_send_datagram` call.